File: 10811/10811.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10811.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/04/23 11:35:41 by jihoolee          #+#    #+#             */
/*   Updated: 2023/04/23 12:34:33 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>
#include <algorithm>

int main(void) {
  int basket_cnt;
  int move_cnt;
  std::cin >> basket_cnt >> move_cnt;
  std::vector<int>  baskets(basket_cnt);

  for (size_t idx = 0; idx < basket_cnt; ++idx)
    baskets[idx] = idx + 1;
  for (int move_i = 0; move_i < move_cnt; ++move_i) {
    int i, j;

    std::cin >> i >> j;
    std::reverse(baskets.begin() + i - 1, baskets.begin() + j);
  }
  for (std::vector<int>::const_iterator it = baskets.cbegin();
        it != baskets.cend();
        ++it)
    std::cout << *it << ' ';
  return 0;
}

File: 10811/10811.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10811.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2023/04/23 11:37:04 by jihoolee          #+#    #+#              #
#    Updated: 2023/04/23 12:55:22 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main():
    basket_cnt, move_cnt = map(int, input().split())
    baskets = [i + 1 for i in range(basket_cnt)]
    for _ in range(move_cnt):
        i, j = map(int, input().split())
        baskets[i - 1:j] = reversed(baskets[i - 1:j])
    print(*baskets)


if (__name__ == '__main__'):
    main()
